src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void resize() {
+        if (size == elements.length) {
+            T[] newElements = (T[]) new Object[size + size / 2];
+            System.arraycopy(elements, 0, newElements, 0, size);
+            elements = newElements;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid.");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
```

Comment:
Better create a checkSize () method that will check the size and in which case call remove

Reply:
it's realized in the resize method, isn't it?

Reply:
it's better to rename this method(`resize()`) to `ensureCapacity` or smth like this